DataFormats/TrackerRecHit2D/src/VectorHit.cc

```diff
@@ -0,0 +1,413 @@
+  n2[1] = gPositionUpper.y() - gPositionLower.y();
+
+  double n3 = n1[0] * n2[0] + n1[1] * n2[1];
+  double signCurv = -copysign(1.0, n3);
+  double phi1 = atan2(gPositionUpper.y() - gPositionLower.y(), gPositionUpper.x() - gPositionLower.x());
+
+  if (h1 != 0) {
+    double h2 = 2 * h1;
+    double r12 = pow(gPositionLower.x(), 2) + pow(gPositionLower.y(), 2);
+    double r22 = pow(gPositionUpper.x(), 2) + pow(gPositionUpper.y(), 2);
+    double h3 =
+        (pow(gPositionLower.x(), 2) - 2. * gPositionLower.x() * gPositionUpper.x() + pow(gPositionUpper.x(), 2) +
+         pow(gPositionLower.y(), 2) - 2. * gPositionLower.y() * gPositionUpper.y() + pow(gPositionUpper.y(), 2));
+    double h4 = -pow(gPositionLower.x(), 2) * gPositionUpper.x() + gPositionLower.x() * pow(gPositionUpper.x(), 2) +
+                gPositionLower.x() * pow(gPositionUpper.y(), 2) - gPositionUpper.x() * pow(gPositionLower.y(), 2);
+    double h5 = pow(gPositionLower.x(), 2) * gPositionUpper.y() - pow(gPositionUpper.x(), 2) * gPositionLower.y() +
+                pow(gPositionLower.y(), 2) * gPositionUpper.y() - gPositionLower.y() * pow(gPositionUpper.y(), 2);
+
+    //radius of circle
+    double invRho2 = (4. * h1 * h1) / (r12 * r22 * h3);
+    curvature = sqrt(invRho2);
+
+    //center of circle
+    double xcentre = h5 / h2;
+    double ycentre = h4 / h2;
+
+    //to compute phi at the cluster points
+    double xtg = gPositionLower.y() - ycentre;
+    double ytg = -(gPositionLower.x() - xcentre);
+
+    //to compute phi at the origin
+    phi = atan2(ytg, xtg);
+
+    AlgebraicROOTObject<4, 4>::Matrix jacobian;
+    for (int i = 0; i < 4; i++) {
+      for (int j = 0; j < 4; j++) {
+        jacobian[i][j] = 0.0;
+      }
+    }
+    double denom1 = 1. / sqrt(r12 * r22 * h3);
+    double denom2 = 1. / (pow(r12 * r22 * h3, 1.5));
+    jacobian[0][0] = 1.0;  // dx1/dx1 dx1/dy1 dx2/dx1 dy2/dx1
+    jacobian[1][1] = 1.0;  //dy1/dx1 dy1/dy1 dy2/dx1 dy2/dx1
+    jacobian[2][0] =
+        (h1 * (2. * gPositionLower.x() * r22 * h3 + (2. * gPositionLower.x() - 2. * gPositionUpper.x()) * r12 * r22)) *
+            denom2 -
+        (2. * gPositionUpper.y()) * denom1;  // dkappa/dx1
+    jacobian[2][1] =
+        (2. * gPositionUpper.x()) * denom1 +
+        (h1 * (2. * gPositionLower.y() * r22 * h3 + r12 * r22 * (2. * gPositionLower.y() - 2. * gPositionUpper.y()))) *
+            denom2;  // dkappa/dy1
+    jacobian[2][2] =
+        (2. * gPositionLower.y()) * denom1 +
+        (h1 * (2. * gPositionUpper.x() * r12 * h3 - 2. * (gPositionLower.x() - gPositionUpper.x()) * r12 * r22)) *
+            denom2;  // dkappa/dx2
+    jacobian[2][3] =
+        (h1 * (2. * gPositionUpper.y() * r12 * h3 - r12 * r22 * 2. * (gPositionLower.y() - gPositionUpper.y()))) *
+            denom2 -
+        (2. * gPositionLower.x()) * denom1;  // dkappa/dy2
+
+    for (int i = 0; i < 4; i++) {
+      jacobian[2][i] = -jacobian[2][i];
+    }
+
+    AlgebraicVector2 M;
```

Comment:
both single-letter and capitalized local vars are against the CMSSW style

Reply:
Fixed by expanding variable names.